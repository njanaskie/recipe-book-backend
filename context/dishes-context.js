import React from 'react';

const DishesContext = React.createContext()

export { DishesContext as default }