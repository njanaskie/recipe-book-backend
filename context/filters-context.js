import React, { useContext, useReducer } from 'react';
import filtersReducer, { filtersReducerDefaultState } from '../src/reducers/filters'

const FiltersContext = React.createContext()

export const useFiltersContext = () => useContext(FiltersContext)

const FiltersProvider = ({ children }) => {
    const [filters, filtersDispatch] = useReducer(filtersReducer, filtersReducerDefaultState)

    return (
        <FiltersContext.Provider value={{ filters, filtersDispatch }}>
            {children}
        </FiltersContext.Provider>
    )
}

export { FiltersProvider, FiltersContext as default }