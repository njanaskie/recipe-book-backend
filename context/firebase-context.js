import React, { useEffect, useState, useContext, useReducer } from 'react';
import database, { firebase, googleAuthProvider } from '../src/firebase/firebase';
import { history } from '../src/routers/AppRouter';
import authReducer from '../src/reducers/auth'
import LoadingPage from '../src/components/LoadingPage'

const FirebaseContext = React.createContext()

export const useFirebaseContext = () => useContext(FirebaseContext)

const FirebaseProvider = ({ children }) => {
    const [auth, authDispatch] = useReducer(authReducer, [])
    // const authDispatch = useDispatch()
    const [user, setUser] = useState('')
    const [loading, setLoading] = useState(true)
    const [isAdmin, setIsAdmin] = useState()
    const [isGuest, setIsGuest] = useState()
    const email = process.env.GUEST_EMAIL
    const password = process.env.GUEST_PASSWORD

    const login = () => {
        firebase.auth().signInWithPopup(googleAuthProvider).then(() => {
            authDispatch({ type: 'LOGIN' })
        })
    }

    const loginAsGuest = () => {
        firebase.auth().signInWithEmailAndPassword(email, password).then(() => {
            authDispatch({ type: 'LOGIN_AS_GUEST' })
        })
    }

    const logout = () => {
        firebase.auth().signOut().then(() => {
            authDispatch({ type: 'LOGOUT' })
        });
    }

    useEffect(() => {
        const unsubscribe = firebase.auth().onAuthStateChanged((user) => {
            setUser(user)
            setLoading(false)
            setIsAdmin()
            setIsGuest()
        })

        return () => unsubscribe()
    }, [])

    useEffect(() => {
        if (user) {
            database.collection('users').doc(user.uid).get()
            .then((doc) => {
                const userData = doc.data()
                if (userData) {
                    if (userData.isAdmin === true) {
                        const isAdmin = userData.isAdmin
                        setIsAdmin(isAdmin)
                    } else if (userData.isGuest === true) {
                        const isGuest = userData.isGuest
                        setIsGuest(isGuest)
                    }
                }
            })
        }
    }, [user])

    if (loading) {
        return <LoadingPage />
    }
    // if (user) {
    //     if (history.location.pathname === '/') {
    //         history.push('/dashboard')
    //     }
    // } else {
    //     history.push('/');
    // }

    return (
        <FirebaseContext.Provider value={{ user, isAdmin, isGuest, login, loginAsGuest, logout }}>
            {children}
        </FirebaseContext.Provider>
    )
}

export { FirebaseProvider, FirebaseContext as default }