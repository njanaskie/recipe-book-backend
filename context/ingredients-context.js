import React from 'react';

const IngredientsContext = React.createContext()

export { IngredientsContext as default }