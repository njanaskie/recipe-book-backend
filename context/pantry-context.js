import React, { useContext } from 'react';

const PantryContext = React.createContext()

export const usePantryContext = () => useContext(PantryContext)

export { PantryContext as default }