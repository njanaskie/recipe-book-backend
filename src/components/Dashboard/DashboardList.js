// import React, { useCallback, useState, useEffect, useMemo, useContext, useRef } from 'react'
// import DishesContext from '../../../context/dishes-context'
// import DishListItem from '../Dishes/DishListItem'
// import useDishes from '../../hooks/useDishes'
// import usePantryIngredients from '../../hooks/usePantryIngredients'
// import usePantryDishes from '../../hooks/usePantryDishes'
// import { Input } from 'semantic-ui-react'
// import DashboardListFilters from './DashboardListFilters'
// import { useFiltersContext } from '../../../context/filters-context'
// import pantryDishNamesTest from '../../selectors/pantry-dishes'

// export const DashboardList = () => {
//     const { filters } = useFiltersContext()
//     // const dishes = useDishes()
//     // const pantryIngredients = usePantryIngredients()
//     // const pantryDishNames = pantryDishNamesTest(pantryIngredients, dishes)
//     // const pantryDishes = usePantryDishes()
    
//     return (
//         <div className='content-container'>
//             <DashboardListFilters filters={filters}/>
//             <div className='dish-table'>
//                 {
//                     pantryDishes.length === 0 ? (
//                         <div>
//                             <span>No dishes</span>
//                         </div>
//                     ) : (
//                         pantryDishes.map((dish) => 
//                             <DishListItem key={dish.id} dish={dish} />
//                         )
//                     )
//                 }
//             </div>
//         </div>
//     )
// }

// export default DashboardList