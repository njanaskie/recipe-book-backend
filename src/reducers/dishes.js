
// Dishes Reducer

const dishesReducerDefaultState = [];

const dishesReducer = (state = dishesReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_DISH':
            return [
                ...state,
                action.dish
            ];
        case 'REMOVE_DISH':
            return state.filter(({ id }) => id !== action.id);
        case 'EDIT_DISH':
            return state.map((dish) => {
                if (dish.id === action.id) {
                    return {
                        ...dish,
                        ...action.updates
                    };
                } else {
                    return dish;
                };
            });
        case 'SET_DISHES':
            return action.dishes;
        default:
            return state;
    }
}

export default dishesReducer;