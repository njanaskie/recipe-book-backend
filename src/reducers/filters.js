
// Filters Reducer

const filtersReducerDefaultState = {
    text: '',
    keyIngredients: [],
};

const filtersReducer = (state = filtersReducerDefaultState, action) => {
    switch (action.type) {
        case 'SET_TEXT_FILTER':
            return {
                ...state,
                text: action.text
            }
        case 'SET_KEY_INGREDIENT_FILTER':
            return {
                ...state,
                keyIngredients: action.keyIngredients
            }
        default:
            return state;
        }
    }
    
    export default filtersReducer;