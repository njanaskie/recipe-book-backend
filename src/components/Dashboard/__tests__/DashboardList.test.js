import * as ReactAll from 'react'
import React from 'react'
import { shallow, mount } from 'enzyme'
import { DashboardList } from '../DashboardList'
import * as AllFiltersContext from '../../../../context/filters-context'
import FiltersContext, { useFiltersContext } from '../../../../context/filters-context'
import * as AllDishesContext from '../../../../context/dishes-context'
import DishesContext, { useDishesContext } from '../../../../context/dishes-context'
import * as AllFirebaseContext from '../../../../context/firebase-context'
import PantryContext from '../../../../context/pantry-context'
import { filters } from '../../../tests/fixtures/filters'
import dishes from '../../../tests/fixtures/dishes'

test('should render DashboardList with dishes',() => {
    const wrapper = shallow(
        <DishesContext.Provider value={dishes}>
            <FiltersContext.Provider value={filters}>
                <DashboardList dishes={dishes}/>
            </FiltersContext.Provider>
        </DishesContext.Provider>
        )
    expect(wrapper).toMatchSnapshot()
});


test('should render DashboardList with no dishes',() => {
    // jest.spyOn(AllFiltersContext, 'useFiltersContext').mockImplementation(() => filters)
    // jest.spyOn(AllDishesContext, 'useDishesContext').mockImplementation(() => dishes)
    const wrapper = shallow(
        <FiltersContext.Provider value={[]}>
            <DishesContext.Provider value={{}}>
                <DashboardList dishes={[]}/>
            </DishesContext.Provider>
        </FiltersContext.Provider>
        )
    expect(wrapper).toMatchSnapshot()
});
