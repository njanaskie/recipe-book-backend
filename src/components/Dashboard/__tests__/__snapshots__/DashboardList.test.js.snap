// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DashboardList with dishes 1`] = `
<ContextProvider
  value={
    Object {
      "cuisine": "",
      "dishType": "",
      "keyIngredients": Array [],
      "text": "",
    }
  }
>
  <DashboardList
    dishes={
      Array [
        Object {
          "cuisine": "dinner",
          "description": "test",
          "id": "1",
          "keyIngredients": Array [
            "meatballs",
            "roll",
            "marinara sauce",
          ],
          "name": "Meatball Sub",
          "optionalIngredients": Array [
            "parmesan",
          ],
          "recipes": Array [],
          "type": "",
        },
        Object {
          "cuisine": "snack",
          "description": "test 2",
          "id": "2",
          "keyIngredients": Array [
            "bread",
            "butter",
          ],
          "name": "Butter Toast",
          "optionalIngredients": Array [],
          "recipes": Array [],
          "type": "",
        },
        Object {
          "cuisine": "dinner",
          "description": "test 3",
          "id": "3",
          "keyIngredients": Array [
            "perogies",
          ],
          "name": "Perogies",
          "optionalIngredients": Array [
            "spinach",
            "chicken",
            "garlic",
          ],
          "recipes": Array [],
          "type": "",
        },
      ]
    }
  />
</ContextProvider>
`;

exports[`should render DashboardList with no dishes 1`] = `
<ContextProvider
  value={Object {}}
>
  <DashboardList
    dishes={Array []}
  />
</ContextProvider>
`;
